gear-test/src/check.rs

```diff
@@ -423,25 +424,27 @@ pub fn check_memory(
 }
 
 fn check_active_programs(
-    expected_ids: Vec<ProgramId>,
-    actual_ids: Vec<ProgramId>,
+    expected_programs: Vec<(ProgramId, bool)>,
+    actual_programs: &BTreeMap<ProgramId, bool>,
+    exact: bool,
 ) -> Result<(), Vec<String>> {
-    let mut errors = Vec::with_capacity(expected_ids.len());
-    if expected_ids.len() != actual_ids.len() {
+    let mut errors = Vec::new();
+
+    if exact && expected_programs.len() != actual_programs.len() {
         errors.push(format!(
-            "invalid amount of active programs: expected - {:?}, actual - {:?}",
-            expected_ids.len(),
-            actual_ids.len()
+            "invalid amount of programs: expected - {:?}, actual - {:?}",
+            expected_programs.len(),
+            actual_programs.len()
         ));
-    } else {
-        let check_data = expected_ids.iter().zip(actual_ids.iter());
-        for (idx, (expected_id, actual_id)) in check_data.enumerate() {
-            if expected_id != actual_id {
-                errors.push(format!(
-                    "invalid program id at position {:?}. Expected - {:?}, actual - {:?}",
-                    idx, expected_id, actual_id
-                ));
-            }
+    }
+
+    for (id, terminated) in expected_programs {
+        let actual_termination = actual_programs.get(&id);
+        if actual_termination != Some(&terminated) {
+            errors.push(format!(
+                "invalid program id. {:?} expected to be {:?}, but it is {:?}",
+                id, terminated, actual_termination,
```

Comment:
better to output here stringed bools

Reply:
I write a more clear error message there, when we can't find program
done